_rules/text-contrast-widget-nqzcj8.md

```diff
@@ -0,0 +1,443 @@
+---
+id: nqzcj8
+name: Text inside widget has minimum contrast
+rule_type: atomic
+description: |
+  This rule checks that, for text in widgets, the highest possible contrast of every text character with its background meets the minimal contrast requirement.
+accessibility_requirements:
+  wcag20:1.4.3: # Contrast (Minimum)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+  wcag20:1.4.6: # Contrast (Enhanced)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - Accessibility Tree
+  - DOM Tree
+  - CSS Styling
+  - Language
+acknowledgments:
+  authors:
+    - Jean-Yves Moyen
+---
+
+## Applicability
+
+This rule applies to triplet of an HTML element, a character in a [text node][], and a set of [widget pseudo-classes][] for which all the following are true:
```

Comment:
You're not giving the relationship between the element and the text node. Why not apply this to any visible text node that is a descendant of a widget?

Reply:
I don't get it.
The [**ancestor**](https://github.com/act-rules/act-rules.github.io/pull/1455/files#diff-b91b63566cb2236e0fc26f885035436226b9a4acae9e9ae534c66ac806d4572bR33) condition is giving that relationship.

I felt it was a bit easier to explicitly mention the element here as it allows easier references to it in the conditions below.

Reply:
Ahh I see... okay I find that confusing.